### PR TITLE
Laboratory work #1, Semenova Taisiya - 23FPL2

### PR DESCRIPTION
**Pull request by semtassta**
Adding print

### CODE REVIEW

File: seminars/practice_1_run_debug.py [hunk 1]
```diff
@@ -4,6 +4,9 @@
 Running python application and debugging
 """
 
+
```
[the remark this answers is not in the document]
Comment (semtassta) [reply seminars/practice_1_run_debug.py:7]: done

File: lab_1_classify_by_unigrams/start.py [hunk 2]
```diff
@@ -14,8 +21,20 @@ def main() -> None:
         de_text = file_to_read_de.read()
     with open("assets/texts/unknown.txt", "r", encoding="utf-8") as file_to_read_unk:
         unknown_text = file_to_read_unk.read()
-    result = None
+    result = 1  # None
+    print(tokenize(en_text))
```
[the remark this answers is not in the document]
Comment (semtassta) [reply lab_1_classify_by_unigrams/start.py:25]: done

File: lab_1_classify_by_unigrams/main.py [hunk 3]
```diff
@@ -152,7 +250,7 @@ def collect_profiles(paths_to_profiles: list) -> list[dict[str, str | dict[str,
 
 
 def detect_language_advanced(
-    unknown_profile: dict[str, str | dict[str, float]], known_profiles: list
+        unknown_profile: dict[str, str | dict[str, float]], known_profiles: list
```
[the remark this answers is not in the document]
Comment (semtassta) [reply lab_1_classify_by_unigrams/main.py:253]: done

File: lab_1_classify_by_unigrams/main.py [hunk 4]
```diff
@@ -119,6 +211,12 @@ def load_profile(path_to_file: str) -> dict | None:
 
     In case of corrupt input arguments, None is returned
     """
+    path_to_file_components = path_to_file.split('/')
+    file_name_and_extension = path_to_file_components[-1].rsplit('.', 1)
+    with open(path_to_file, 'r') as profile_file:
+        profile = {'name': file_name_and_extension[0],
```
[the remark this answers is not in the document]
Comment (semtassta) [reply lab_1_classify_by_unigrams/main.py:217]: done

File: lab_1_classify_by_unigrams/main.py [hunk 5]
```diff
@@ -84,12 +118,52 @@ def compare_profiles(
...
+            elif letter in compare_letters_list:
+                compare_frequencies_list.append(profile_to_compare['freq'][letter])
+                unknown_frequencies_list.append(0)
+            else:
+                unknown_frequencies_list.append(unknown_profile['freq'][letter])
+                compare_frequencies_list.append(0)
+        return calculate_mse(compare_frequencies_list, unknown_frequencies_list)
+    else:
+        return None
+    # if (isinstance(unknown_profile, dict) and 'name' in unknown_profile and 'freq' in unknown_profile and
```
[the remark this answers is not in the document]
Comment (semtassta) [reply lab_1_classify_by_unigrams/main.py:141]: done

File: lab_1_classify_by_unigrams/main.py [hunk 6]
```diff
@@ -84,12 +118,52 @@ def compare_profiles(
     In case of corrupt input arguments or lack of keys 'name' and
     'freq' in arguments, None is returned
     """
+    if (isinstance(unknown_profile, dict) and 'name' in unknown_profile and 'freq' in unknown_profile and
```
[the remark this answers is not in the document]
Comment (semtassta) [reply lab_1_classify_by_unigrams/main.py:121]: done

File: lab_1_classify_by_unigrams/main.py [hunk 7]
```diff
@@ -84,12 +118,52 @@ def compare_profiles(
...
+    #     return round(calculate_mse(compare_frequencies_list, unknown_frequencies_list), 3)
+    # else:
+    #     return None
 
 
 def detect_language(
-    unknown_profile: dict[str, str | dict[str, float]],
-    profile_1: dict[str, str | dict[str, float]],
-    profile_2: dict[str, str | dict[str, float]],
+        unknown_profile: dict[str, str | dict[str, float]],
```
[the remark this answers is not in the document]
Comment (semtassta) [reply lab_1_classify_by_unigrams/main.py:164]: done

File: lab_1_classify_by_unigrams/main.py [hunk 8]
```diff
@@ -105,6 +179,24 @@ def detect_language(
 
     In case of corrupt input arguments, None is returned
     """
+    if isinstance(unknown_profile, dict) and isinstance(profile_1, dict) and isinstance(profile_2, dict):
```
[the remark this answers is not in the document]
Comment (semtassta) [reply lab_1_classify_by_unigrams/main.py:182]: done

File: lab_1_classify_by_unigrams/main.py [hunk 9]
```diff
@@ -119,6 +211,12 @@ def load_profile(path_to_file: str) -> dict | None:
 
     In case of corrupt input arguments, None is returned
     """
+    path_to_file_components = path_to_file.split('/')
+    file_name_and_extension = path_to_file_components[-1].rsplit('.', 1)
+    with open(path_to_file, 'r') as profile_file:
```
[the remark this answers is not in the document]
Comment (semtassta) [reply lab_1_classify_by_unigrams/main.py:216]: Excuse me, I couldn't quite understand exactly what you want me to change here. I'm trying to use the open() function like it's demonstrated here: https://docs.python.org/3/library/functions.html#open. And I couldn't find the method path.open(). Would you please provide me with further instructions on the topic?

**Review by semtassta — COMMENTED** [review]
image
how can I pass those MyPy checks? how can I adjust the types of values?

File: lab_1_classify_by_unigrams/main.py [hunk 10]
```diff
@@ -20,6 +22,13 @@ def tokenize(text: str) -> list[str] | None:
 
     In case of corrupt input arguments, None is returned
     """
+    if not isinstance(text, str):
+        return None
+    list_of_letters = []
+    for character in text:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:28]: better to call `.lower()` for full string

File: lab_1_classify_by_unigrams/main.py [hunk 11]
```diff
@@ -64,11 +83,20 @@ def calculate_mse(predicted: list, actual: list) -> float | None:
...
+    for i in range(n):
+        mse += ((actual[i] - predicted[i]) ** 2)
+    mse /= n
+    return mse
 
 
 def compare_profiles(
-    unknown_profile: dict[str, str | dict[str, float]],
-    profile_to_compare: dict[str, str | dict[str, float]],
+        unknown_profile: dict[str, str | dict[str, float]],
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:98]: revert

File: lab_1_classify_by_unigrams/main.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (artyomtugaryov) [reply lab_1_classify_by_unigrams/main.py:220]: again - no

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/main.py [hunk 12]
```diff
@@ -64,11 +83,20 @@ def calculate_mse(predicted: list, actual: list) -> float | None:
 
     In case of corrupt input arguments, None is returned
     """
+    if (not isinstance(predicted, list) or not isinstance(actual, list) or
+            not len(actual) == len(predicted)):
+        return None
+    n = len(actual)
```
[the remark this answers is not in the document]
Comment (semtassta) [reply lab_1_classify_by_unigrams/main.py:93]: done

File: lab_1_classify_by_unigrams/main.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (semtassta) [reply lab_1_classify_by_unigrams/main.py]: done

File: lab_1_classify_by_unigrams/main.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (semtassta) [reply lab_1_classify_by_unigrams/main.py]: done

File: seminars/practice_2_strings.py [hunk 13]
```diff
@@ -188,6 +187,8 @@ def string_match(first_string: str, second_string: str) -> int:
     """
     # student realisation goes here
 
+
```
[the remark this answers is not in the document]
Comment (semtassta) [reply seminars/practice_2_strings.py]: done

File: seminars/practice_1_run_debug.py [hunk 14]
```diff
@@ -32,7 +32,6 @@
 third_num = first_num / second_num
 print(f'Numbers: {first_num}, {second_num} and {third_num}')
 
-
```
[the remark this answers is not in the document]
Comment (semtassta) [reply seminars/practice_1_run_debug.py]: done

File: lab_1_classify_by_unigrams/main.py (hunk 11 above)
Comment (semtassta) [reply lab_1_classify_by_unigrams/main.py]: done

File: lab_1_classify_by_unigrams/main.py (hunk 10 above)
Comment (semtassta) [reply lab_1_classify_by_unigrams/main.py]: done

**Review by artyomtugaryov — APPROVED** [review]
Looks good

**Pull request closed without merging**